src/main/java/bio/terra/service/snapshot/flight/create/NotifyUserOfSnapshotCreationStep.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class NotifyUserOfSnapshotCreationStep extends DefaultUndoStep {
+  private final SnapshotBuilderService snapshotBuilderService;
+  private final AuthenticatedUserRequest user;
+  private final UUID snapshotRequestId;
+
+  public NotifyUserOfSnapshotCreationStep(
+      SnapshotBuilderService snapshotBuilderService,
+      AuthenticatedUserRequest user,
+      UUID snapshotRequestId) {
+    this.snapshotBuilderService = snapshotBuilderService;
+    this.user = user;
+    this.snapshotRequestId = snapshotRequestId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    snapshotBuilderService.notifySnapshotReady(user, snapshotRequestId);
+    return StepResult.getStepResultSuccess();
```

Comment:
When does this method get called and why does it return StepResult.getStepResultSuccess()?

Reply:
Steps in a flight are executed in the order they were added, so to see this step in its flight context, see where the constructor for the class is being called in the class `SnapshotCreateFlight`.

A step can return different statuses depending on if it failed, or can be retried, or other states (see `StepStatus` for the different possible return values). In this case I return `StepResult.getStepResultSuccess()` to indicate that the step completed successfully and can be marked as complete.